### PR TITLE
oidc: allow non-GSuite OIDC providers from Google

### PR DESCRIPTION
**Pull request by awly**
You can follow
https://developers.google.com/identity/protocols/oauth2/openid-connect
to set up an OIDC provider in GCP, which is distinct from GSuite.

This provider uses the same issuer URL ("https://accounts.google.com")
as GSuite. Our code assumes that if that issuer URL is set, then it must
be GSuite, which is not correct. Only attempt to pull more data from
GSuite API if `google_service_account_uri` is set.

Note: this is not a v6 regression. This issue was introduced back in Nov 2019.

### CODE REVIEW

**Review by webvictim — APPROVED** [review]

**Review by benarent — APPROVED** [review]
( note to remind myself to update the video for Workspace after this is updated )

**Comment by stevenGravy** [comment]
@awly  the `google_service_account` changes are in.  Can we change this to handle the other field too?

**Comment by awly** [comment]
@stevenGravy updated

**Comment by awly** [comment]
Ping @russjones for approval

File: lib/auth/oidc.go [hunk 1]
```diff
@@ -829,9 +829,13 @@ func (a *Server) getClaims(oidcClient *oidc.Client, connector services.OIDCConne
 		return nil, trace.Wrap(err)
 	}
 
-	// for GSuite users, fetch extra data from the proprietary google API
-	// only if scope includes admin groups readonly scope
-	if connector.GetIssuerURL() == teleport.GSuiteIssuerURL {
+	// For GSuite users, fetch extra data from the proprietary google API.
```
Comment (benarent) [inline lib/auth/oidc.go:832]:
```suggestion
	// For Google Workspace users, fetch extra data from the proprietary Google groups API.
```

File: lib/auth/oidc.go [hunk 2]
```diff
@@ -829,9 +829,13 @@ func (a *Server) getClaims(oidcClient *oidc.Client, connector services.OIDCConne
...
+	//
+	// If google_service_account_uri and google_service_account are not set, we
+	// assume that this is a non-GSuite Google OIDC provider using the same
```
Comment (benarent) [inline lib/auth/oidc.go:835]:
```suggestion
	// assume that this is a non-GWorkspace OIDC provider using the same
```

File: lib/auth/oidc.go [hunk 3]
```diff
@@ -829,9 +829,13 @@ func (a *Server) getClaims(oidcClient *oidc.Client, connector services.OIDCConne
...
+	// issuer URL as GSuite (e.g.
```
Comment (benarent) [inline lib/auth/oidc.go:836]:
```suggestion
	// issuer URL as Google Workspace (e.g.
```

**Comment by awly** [comment]
@russjones please review

**Review by russjones — APPROVED** [review]

**Pull request merged**